dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java

```diff
@@ -412,41 +413,47 @@ public Host find(final Contentlet contentlet,
      */
     @Override
     @WrapInTransaction
-    public Host save(Host host, User user, boolean respectFrontendRoles) throws DotSecurityException, DotDataException {
-        if(host != null){
-            hostCache.remove(host);
+    public Host save(final Host hostToBeSaved, User user, boolean respectFrontendRoles) throws DotSecurityException, DotDataException {
+        if(hostToBeSaved != null){
+            hostCache.remove(hostToBeSaved);
         }
 
         Contentlet contentletHost;
         try {
-            contentletHost = APILocator.getContentletAPI().checkout(host.getInode(), user, respectFrontendRoles);
+            contentletHost = APILocator.getContentletAPI().checkout(hostToBeSaved.getInode(), user, respectFrontendRoles);
         } catch (DotContentletStateException e) {
 
             contentletHost = new Contentlet();
             contentletHost.setStructureInode(hostType().inode() );
         }
 
-        contentletHost.getMap().put(Contentlet.DONT_VALIDATE_ME, host.getMap().get(Contentlet.DONT_VALIDATE_ME));
-        APILocator.getContentletAPI().copyProperties(contentletHost, host.getMap());
+        if(null != contentletHost.get(Host.HOST_NAME_KEY) && null != hostToBeSaved.get(Host.HOST_NAME_KEY) &&
+                !contentletHost.get(Host.HOST_NAME_KEY).equals(hostToBeSaved.get(Host.HOST_NAME_KEY)) &&
+                !hostToBeSaved.getBoolProperty("forceExecution")){
+            throw new BadRequestException("Updating the hostName is a Dangerous Execution, to achieve this 'forceExecution': true property needs to be sent.");
```

Comment:
BadRequestException is an exception only for rest level, at API level would be more convenient:
IllegalArgumentException